app/scripts/controllers/decrypt-message.ts

```diff
@@ -35,6 +36,27 @@ const stateMetadata = {
   unapprovedDecryptMsgCount: { persist: false, anonymous: false },
 };
 
+export const isEIP1024EncryptedMessage = (message: {
+  from: string;
+  data: unknown;
+}): message is {
+  from: string;
+  data: Eip1024EncryptedData;
+} => {
+  if (
+    message.data &&
+    typeof message.data === 'object' &&
```

Comment:
Nit: Might be nice to use the `isObject` utility function here instead. What you have here is equivalent, but `isObject` is a bit easier to understand (readers might not understand that the truthy check is meant to rule out `null` which is also `typeof object`).
```suggestion
    isObject(message.data) && 
```